src/loading-indicator/index.tsx

```diff
@@ -6,29 +6,113 @@ import { create, tsx } from '@dojo/framework/core/vdom';
 export interface LoadingIndicatorProperties extends ThemedProperties {
 	/** If the element is actively loading. Defaults to true */
 	active?: boolean;
+	/* Sets loader type to linear or circular */
+	type?: 'linear' | 'circular-small' | 'circular-medium' | 'circular-large';
 }
 
+const typeExp = /^([a-zA-Z]+)(?:[\-]{1})([a-zA-Z]+$)/;
+
 const factory = create({ theme }).properties<LoadingIndicatorProperties>();
 
 export const LoadingIndicator = factory(function LoadingIndicator({
 	properties,
 	middleware: { theme }
 }) {
 	const classes = theme.classes(css);
-	const { active = true } = properties();
+	const { active = true, type = 'linear' } = properties();
+
+	const [loadingType, size = 'small'] =
```

Comment:
Can we not do this, it's better to be more explicit/verbose 😄 

```tsx
const isLinear = type === 'linear';

if (isLinear) {
    return (
        // the existing loading indicator
    )
}

let sizeClass: string | undefined;
switch (type) {
	case 'circular-small':
		sizeClass = themeCss.small;
		break;
	case 'circular-large':
		spacingClass = themeCss.large;
		break;
	case 'circular-medium':
	default:
		sizeClass = themeCss.medium;
		break;
}

return (
    // The circular indicator using the sizeClass
)
```

Reply:
Yea I was planning ahead too much, figured with that code we could add more types like `spin-small` without having to update the code. However over engineering for the future is thing